dimagi/commcare-hq
# Web Apps accessibility fixes for expandable groups, pagination and case list search

## Summary
Continue work to improve Web Apps accessibility. The overall project is described in [this document](https://docs.google.com/document/d/1fHNaVlaFl1xrbtPbMqj0YpByss1e0_hzCRlAUbVLboM/edit?usp=sharing). This PR continues to improve navigation within Web Apps for screen reader users and also fixes some issues found in QA testing.

### Related issues:
https://dimagi-dev.atlassian.net/browse/USH-499
https://dimagi-dev.atlassian.net/browse/USH-500

## Product Description

1. Collapsible groups (i.e. groups with the advanced attribute "group-collapse") are added to the tab order so they become focusable. They have ARIA labels and they activate on either a Space or Enter key press. This is largely based on the [ARIA accordion pattern](https://www.w3.org/TR/wai-aria-practices/#accordion).
2. Pagination links behave like accessible links. They are added to the tab order so they become focusable. They have labels and activate on an Enter key press.
3. When Remote Case Search is enabled, the Search All Cases button activates on either a Space or Enter key press. This change was actually made by changing CaseListView so that a smaller area (just the search text box and button) triggers the keyAction function. It works because the default behavior of an HTML button maps Space and Enter to a click event.
4. On the Manage Organization Structure page, the X button to clear the location field has an ARIA label. This issue was reported in our accessibility vendor's review.

### QA Plan
The QA plan will be to check navigation of these pages by mouse to ensure there are no regressions and to test navigation using a screen reader to see if the changes actually work.

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-2245)

### Safety story
WAI-ARIA attributes are designed to have no effect outside of accessibility tools, so they shouldn't cause any regressions.

The other changes are putting elements into the focus ring and handling key clicks on them, which should not affect existing users who navigate by mouse clicks.

### Rollback instructions
